smack-experimental/src/main/java/org/jivesoftware/smackx/muclight/MultiUserChatLight.java

```diff
@@ -0,0 +1,555 @@
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "MUC Light: " + room + "(" + connection.getUser() + ")";
+    }
+
+    /**
+     * Create new MUCLight.
+     * 
+     * @param roomName
+     * @param subject
+     * @param customConfigs
+     * @param occupants
+     * @return true if it was successfully created
+     * @throws NotConnectedException
+     * @throws XmppStringprepException
+     * @throws InterruptedException
+     */
+    public boolean create(String roomName, String subject, HashMap<String, String> customConfigs, List<Jid> occupants)
+            throws NotConnectedException, XmppStringprepException, InterruptedException {
+        MUCLightCreateIQ createMUCLightIQ = new MUCLightCreateIQ(room, roomName, occupants);
+
+        connection.addSyncStanzaListener(messageListener, fromRoomGroupchatFilter);
+        messageCollector = connection.createPacketCollector(fromRoomGroupchatFilter);
+
+        StanzaFilter responseFilter = new AndFilter(FromMatchesFilter.createBare(room.asDomainBareJid()),
+                new StanzaTypeFilter(IQ.class));
+
+        try {
+            IQ responseIq = connection.createPacketCollectorAndSend(responseFilter, createMUCLightIQ)
+                    .nextResultOrThrow();
+            return responseIq.getType().equals(IQ.Type.result);
+        } catch (NoResponseException | XMPPErrorException e) {
+            removeConnectionCallbacks();
+            return false;
+        } catch (InterruptedException e) {
+            removeConnectionCallbacks();
+            return false;
+        }
+    }
+
+    /**
+     * Create new MUCLight.
+     * 
+     * @param roomName
+     * @param occupants
+     * @return true if it was successfully created
+     * @throws NotConnectedException
+     * @throws XmppStringprepException
+     * @throws InterruptedException
+     */
+    public boolean create(String roomName, List<Jid> occupants)
+            throws NotConnectedException, XmppStringprepException, InterruptedException {
+        return create(roomName, null, null, occupants);
+    }
+
+    /**
+     * Leave the MUCLight.
+     * 
+     * @return true if it was successfully left.
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     */
+    public boolean leave() throws NotConnectedException, InterruptedException, NoResponseException, XMPPErrorException {
+        HashMap<Jid, MUCLightAffiliation> affiliations = new HashMap<>();
+        affiliations.put(connection.getUser(), MUCLightAffiliation.none);
+
+        MUCLightChangeAffiliationsIQ changeAffiliationsIQ = new MUCLightChangeAffiliationsIQ(room, affiliations);
+        StanzaFilter responseFilter = new AndFilter(FromMatchesFilter.createBare(room.asDomainBareJid()),
+                new StanzaTypeFilter(IQ.class));
+        IQ responseIq = connection.createPacketCollectorAndSend(responseFilter, changeAffiliationsIQ)
```

Comment:
Some issues as above. Simply use the IQ request/response idiom and reduce the LOC for the greater good.